### DISCUSSION TITLE
How to run composer install / update  in Herd ?

### DISCUSSION THREAD

**Discussion by magic-thomas** [beyondcode/herd-community#572] [Q&A] [answered]
Hi . I am using php8.2 on M1 Macbook laptop. 

I have another projects which are based on 8.0 and 7.4, but these projects does not `vendor` folder. 

So I have to install packages with composer. How can I do this ?

I tried to find it but all failed to install packages  with old php versions. 

Is it impossible to run 'composer install / update' with another php version in Herd ?

**Comment by mpociot** [comment, marked as answer]
If you have isolated the sites, you can run "herd composer" to use composer with the isolated php version a

**Comment by olehasoft** [comment]
I use Herd on Windows 11, where composer is executed via ~\\.config\\herd\\bin\\composer.bat
```
@ECHO OFF
php "%~dp0composer.phar" %*
```

I copied the file in the same directory to composer81.bat and replaced the startup with the appropriate version of PHP 8.1
```
@ECHO OFF
php81 "%~dp0composer.phar" %*
```
The resulting `composer81` command is faster than isolated `herd composer` and does not require Herd to be running.

I'm sure you can do the same on macOS. It's a shame that Herd doesn't have this by default.
